### DISCUSSION TITLE
empty databases?

### DISCUSSION THREAD

**Discussion by FearWasHere** [karlomikus/bar-assistant#164] [Q&A] [answered]
hi - first, apologies if this is a dumb question.  i'm running via docker-compose, and it seems *almost* all of the databases are empty.  i first noticed it when i tried to log in using the default admin creds, but was unable to; when that didn't work, i just registered a new account.  but it wasn't an admin, so i checked the users table, and sure enough, that account was the only one listed.  (i updated its record, so now i at least have admin.)

anyway, in the web interface, everything except for utensils is completely empty; i've verified this using `sqlite3` from within the container as well.  is this expected?  from the readme & docs, it seems like the databases are supposed to be populated, but maybe i'm missing something.  am i crazy?  and what further info can i provide?  thanks!

**Comment by karlomikus** [comment]
It's possible that you had some issues on first time setup and actions that fill up the bar did not execute.

> Check your logs for error messages. You probably had some problems when setting up the first-time installation, this can lead to skipping initial database data importing.
> 
> Easiest fix is to remove the Bar Assistant container and the related volumes, then recreate the container.

**Comment by FearWasHere** [comment]
thank you - that's what i thought.  in this case, i seem to be having a very strange set of circumstances.  i'll explain, and maybe you (or anyone) can tell me what dumb thing i'm missing.

in order to eliminate variables, i spun a new vm running ubuntu server 22.04lts; i ran all the updates and installed docker/docker-compose.  so far, so good.  i followed the directions here: https://github.com/bar-assistant/docker/blob/master/README.md - the only modifications i made were to set `BASE_URL` and `MEILI_MASTER_KEY` in the .env file, and in the docker-compose file to explicitly set salt-rim to run on 8080, because for whatever reason, hitting the container on 80 just showed the nginx welcome page.

anyway, i'm able to hit salt-rim in a browser, and see that it reports the backend is running.  but...the databases are empty again.  by that, i mean if i exec into the bar-assistant container and issue `sqlite3 /var/www/cocktails/storage/bar-assistant/database.sqlite`, i can see a bunch of tables, but the only one that has any data in it is, for whatever reason, the utensils table.  this is the same behavior i was seeing with my other install, btw, so at the very least i can reproduce it.  

i don't really speak php, but i poked around some and saw that the main function here appears to be completely empty: https://github.com/karlomikus/bar-assistant/blob/9609f1923085de5e30d075e5c9d9bdf1540d66af/database/seeders/DatabaseSeeder.php 

i have no idea if that's related, just something i noticed in a file that sounded likely.  i have, i should mention, also deleted & recreated the containers, both on this vm and my primary docker host.

i feel like i must be missing some silly detail.  any tips for troubleshooting?

thanks!

**Comment by karlomikus** [comment, marked as answer]
What you are looking for is this command: `php artisan bar:open`. This sets up the initial data, but runs only if the database is created for the first time. So if you already have the volume and the database file is created it wont be run. You can see what executes in your container on start here: https://github.com/karlomikus/bar-assistant/blob/master/resources/docker/run.sh . You can try to fix it by running it manually:

```
docker compose exec bar-assistant php artisan bar:open
```

**Comment by FearWasHere** [comment]
brilliant!  that was exactly what i needed - thank you so much!  i can confirm not only that opening the bar worked, but also that this thing is FAST.  looking forward to putting it to good use.  thanks again!
